yetanotherco/aligned_layer
# Add CLI

This PR introduces  a new CLI for the proving system, currently supporting plonk and groth16. 
The CLI is  built using the `urfave /cli/v2` package 

## Type of change

- [x] New 

 ## Usage

To use the CLI for proof verification, users must specify the proof system and provide paths to the necessary files: the proof file, public input file, and verification key file. The paths are resolved relative to the directory of the executable, improving usability across different setups.


### Command Syntax

To run the command, use the following syntax:

```bash
go run task_sender/cmd/main.go --system [SYSTEM] --proof [PROOF_FILE] --public-input [PUBLIC_INPUT_FILE] --verification-key [VERIFICATION_KEY_FILE]
```

### How to Test

Compile the program using Go.
Run the example commands to verify different proof systems.
Attempt using an unsupported proof system to test the error messaging.
